rbac/management/cache.py

```diff
@@ -279,6 +279,36 @@ def set_jwt_response(self, response):
         super().save(self.JWT_CACHE_KEY, response, "JWT response")
 
 
+class PrincipalCache(BasicCache):
+    """Redis-based caching for storing the principals."""
+
+    def key_for(self, tenant: str, principal: str) -> str:
+        return f"rbac::principal::{tenant}::{principal}"
+
+    def set_cache(self, pipe, key, item):
+        """Set cache to redis."""
+        pipe.hset(key, json.dumps(item))
+        pipe.expire(key, settings.PRINCIPAL_CACHE_LIFETIME)
+        pipe.execute()
+
+    def get_from_redis(self, key):
+        """Get principal from redis based on the tenant and the principal."""
+        obj = self.connection.hget(name="", key=key)
+        if obj:
+            return json.loads(obj)
+        else:
+            return None
+
+    def get_principal(self, org_id: str, principal_name: str):
+        super().get_cached(
+            self.key_for(org_id, principal_name),
+            f'[org_id: "{org_id}"][principal_name: "{principal_name}"] Unable to fetch principal from cache',
+        )
+
+    def cache_principal(self, org_id: str, principal_username: str, principal):
```

Comment:
Nit: We can get rid of the principal_username input, and in the code use principal.username